internal/server/middleware.go

```diff
@@ -130,3 +136,28 @@ func loggingMiddleware() func(http.Handler) http.Handler {
 		})
 	}
 }
+
+// httpMetricsMiddleware collects and emits HTTP mode metrics on the first request.
+// Uses sync.Once to ensure metrics are collected exactly once per server session.
+// Requires Basic Auth credentials to be present in the request context (set by basicAuthMiddleware).
```

Comment:
Do we need to check the auth requirement before sending the metrics? Just trying to think what happens if the header is missing/invalid in the first request?